include/ada/url_aggregator-inl.h

```diff
@@ -487,6 +496,12 @@ inline void url_aggregator::clear_base_pathname() {
   buffer.erase(components.pathname_start, pathname_length);
   if (components.search_start != url_components::omitted) { components.search_start -= pathname_length; }
   if (components.hash_start != url_components::omitted) { components.hash_start -= pathname_length; }
+  if(components.pathname_start == components.host_end + 2 && buffer[components.host_end] == '/' && buffer[components.host_end + 1] == '.') {
+    components.pathname_start -= 2;
+    buffer.erase(components.host_end, 2);
+    if (components.search_start != url_components::omitted) { components.search_start -= 2; }
+    if (components.hash_start != url_components::omitted) { components.hash_start -= 2; }
+  }
```

Comment:
We can make this changes after line 495 by just manipulating `pathname_length` attribute. Changing the variable name to something like `difference` can make it much more readable

Reply:
Done.